pkg/codegen/hcl2/model/binder_expression.go

```diff
@@ -769,6 +769,23 @@ func (b *expressionBinder) bindUnaryOpExpression(syntax *hclsyntax.UnaryOpExpr)
 	if tokens == nil {
 		tokens = _syntax.NewUnaryOpTokens(syntax.Op)
 	}
+
+	// simplify UnaryOperation(Negate, Number(N)) into Number(-N)
+	if syntax.Op == hclsyntax.OpNegate {
+		if lit, ok := operand.(*LiteralValueExpression); ok && lit.Value.Type().Equals(cty.Number) {
+			num, _ := lit.Value.AsBigFloat().Int64()
+			negated := &LiteralValueExpression{
+				Value:  cty.NumberIntVal(-num),
```

Comment:
Do we even need to go via int64 for this? Can't we stay as bigFloat given that's what cty is gonna work with internal anyway. Also means this is correct for things like -122.30, which we should probably ensure we have a test for!

Reply:
I think we might be able to just `.Negate()` 🤔 will double check